java/src/jmri/jmrit/display/layoutEditor/LayoutEditorToolBarPanel.java

```diff
@@ -359,6 +362,57 @@ protected void setupComponents() {
         locationPanel.add(yLabel);
         locationPanel.add(new JLabel("}    "));
 
+        locationPopupMenu.add(new AbstractAction(Bundle.getMessage("Pixels")) {
+            @Override
+            public void actionPerformed(ActionEvent event) {
+                locationFormatString = "Pixels";
+                Dimension coordSize = new JLabel("10000").getPreferredSize();
+                xLabel.setPreferredSize(coordSize);
+                yLabel.setPreferredSize(coordSize);
+
+                layoutEditor.redrawPanel();
+            }
+        });
+        locationPopupMenu.add(new AbstractAction(Bundle.getMessage("MetricCM")) {
+            @Override
+            public void actionPerformed(ActionEvent event) {
+                locationFormatString = "MetricCM";
+
+                Dimension coordSize = new JLabel(getMetricCMText(10005)).getPreferredSize();
+                xLabel.setPreferredSize(coordSize);
+                yLabel.setPreferredSize(coordSize);
+
+                layoutEditor.gContext.setGridSize(10);
+                layoutEditor.gContext.setGridSize2nd(10);
+
+                layoutEditor.redrawPanel();
+            }
+        });
+        locationPopupMenu.add(new AbstractAction(Bundle.getMessage("EnglishFeetInches")) {
+            @Override
+            public void actionPerformed(ActionEvent event) {
+                locationFormatString = "EnglishFeetInches";
+
+                Dimension coordSize = new JLabel(getEnglishFeetInchesText(100008)).getPreferredSize();
+                xLabel.setPreferredSize(coordSize);
+                yLabel.setPreferredSize(coordSize);
+
+                layoutEditor.gContext.setGridSize(16);
+                layoutEditor.gContext.setGridSize2nd(12);
+
+                layoutEditor.redrawPanel();
+            }
+        });
+
+        locationPanel.addMouseListener(new MouseAdapter() {
+            @Override
+            public void mousePressed(MouseEvent me) {
+                if (me.isPopupTrigger()) {
+                    locationPopupMenu.show(locationPanel, me.getX(), me.getY());
```

Comment:
Please check for the isPopupTrigger in all three of `mousePressed(MouseEvent e)`, `mouseReleased(MouseEvent e)` and `mouseClicked(MouseEvent e)`.  Different look-and-feels use the mouse differently, so by checking all three you make sure that the popup gesture will be recognize across all types.

Reply:
Done! Good to know! Thanks!